File: Data Structures & Algorithms/minimum-stack/submission-7.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class MinStack:
    def __init__(self):
        self.stack = []
        self.min = None

    def push(self, val: int) -> None:
        if (not self.stack):
            self.stack.append(ListNode(val))
            self.min = val
        else:
            self.stack.append(ListNode(val, self.min))
            if val<self.min:
                self.min = val


    def pop(self) -> None:
        if self.top() == self.min:
            self.min = self.stack[-1].next
        self.stack.pop()
        

    def top(self) -> int:
        return self.stack[-1].val
        

    def getMin(self) -> int:
        return self.min
 

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-7.cpp
using namespace std;
class Solution {
 public:
  int evalRPN(vector<string>& tokens) {
    stack<int> st;
    int a;
    int b;
    string c;
    for (int i = 0; i<tokens.size(); i++){
      c = tokens[i];
      if (c=="+"){
        b = st.top();
        st.pop();
        a = st.top();
        st.pop();
        st.push(a+b);
      }else if (c=="-"){
        b = st.top();
        st.pop();
        a = st.top();
        st.pop();
        st.push(a-b);
      }else if (c=="*"){
        b = st.top();
        st.pop();
        a = st.top();
        st.pop();
        st.push(a*b);
      }else if (c=="/"){
        b = st.top();
        st.pop();
        a = st.top();
        st.pop();
        st.push((int) a/b);
      }else{
        st.push(stoi(c));
      }
    }
    return st.top();
  }
};